### ISSUE TITLE
Modal component that switches to a drawer on mobile

### ISSUE THREAD

**Issue by bartosz-skejcik** [markmead/hyperui#497]
## Behavior
On desktop the component appears as a modal. On mobile it switches into a drawer.
> [!NOTE]
> This option can me toggled

Highly inspired by shadcn

**Comment by tmatheas** [comment]
@bartosz-skejcik Can you explain this issue and assign me this issue ?

**Comment by bartosz-skejcik** [comment]
> @bartosz-skejcik Can you explain this issue and assign me this issue ?

I can't edit the assignees, labels, projects, etc.

This issue is for adding a new component to the current list of components. I noticed that there are no dialog, modal, popup components and the only ones that are in the same category are the [Alerts](https://www.hyperui.dev/components/application-ui/alerts). 
I created this issue after reading the [contribution guide](https://www.hyperui.dev/blog/how-to-contribute) as I want to obviously, contribute to this project.

The looks of the component are mostly similar to the drawer component from [shadcn](https://ui.shadcn.com/docs/components/drawer).

Obviously the content of the drawer will be changes as we don't have the Chart component available. I was thinking I could change the content to a simple form with one input for a task or smth.
Like this:

`Title`: Add new task
`Description`: Fill out the form to add a new task to the current list.
`Input component`:
- `Label`: Task name
`Button`: Confirm
